psycopg_pool/psycopg_pool/pool.py

```diff
@@ -425,11 +425,12 @@ def close(self, timeout: float = 5.0) -> None:
             waiting = list(self._waiting)
             self._waiting.clear()
             connections = list(self._pool)
-            self._pool.clear()
 
-        # Now that the flag _closed is set, getconn will fail immediately,
-        # putconn will just close the returned connection.
-        self._stop_workers(waiting, connections, timeout)
+            # Now that the flag _closed is set, getconn will fail immediately,
+            # putconn will just close the returned connection.
```

Comment:
This comment is now misplaced. "Now" refers to the code out of the `self._lock` block, not to `_stop_workers()`. (my bad, the lack of a blank line made it deceptive).

Reply:
Maybe it'd be best you propose a diff / push the change?

Reply:
Ok, let me try and figure out a different codepath.